lib/bootstrap/bundler.rb

```diff
@@ -176,6 +179,16 @@ def execute_bundler(options)
       ::Bundler::CLI.start(bundler_arguments(options))
     end
 
+    def specific_platform
+      ::Gem.platforms.find {|plat| plat.is_a?(::Gem::Platform) && plat.os=='java' && !plat.cpu.nil?}
+    end
+
+    def genericize_platform
+      output = LogStash::Bundler.invoke!({:add_platform => 'java', :silence_root_warning => true})
+      remove_platform_options = {:remove_platform => specific_platform.to_s, :silence_root_warning => true} unless specific_platform.nil?
+      output << LogStash::Bundler.invoke!(remove_platform_options) unless remove_platform_options.nil?
+    end
```

Comment:
While not possible _directly_ with our tooling as-is, it is possible to end up with a `Gemfile.lock` with more than one specific platform. Since we effectively have a zero-or-more situation, I would prefer to iterate over an enumberable.

```suggestion
    def specific_platforms
      ::Gem.platforms.select {|plat| plat.is_a?(::Gem::Platform) && plat.os=='java' && !plat.cpu.nil?}
    end

    def genericize_platform
      output = LogStash::Bundler.invoke!({:add_platform => 'java', :silence_root_warning => true})
      specific_platform.each do |platform_to_remove|
        output << LogStash::Bundler.invoke!({:remove_platform => platform_to_remove.to_s, :silence_root_warning => true})
      end
    end
```